drivers/platform/mbed/mbed_lvgl.h

```diff
@@ -0,0 +1,87 @@
+ * Copyright (c) 2023 Analog Devices, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *  - Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *  - Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+#ifndef MBED_LVGL_H_
+#define MBED_LVGL_H_
+
+// Platform support needs to be C-compatible to work with other drivers
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+/*****************************************************************************/
+/***************************** Include Files *********************************/
+/*****************************************************************************/
+
+#include "stdint.h"
+
+/******************************************************************************/
+/********************** Variables and User defined data types *****************/
+/******************************************************************************/
+
+/**
+* @struct mbed_lvgl_init_param
+* @brief Structure holding the mbed specific lvgl init parameters
+*/
+struct mbed_lvgl_init_param {
+	uint32_t tick_period_us;	// lvgl tick period in usec
```

Comment:
i'm not familiar with lvgl, but is this tick period lvgl-specific or it's a setting that is only possible when using mbed ?

Reply:
lvgl is a graphics library for portable display based embedded applications. lvgl task handler need one periodic timer with period in msec range. In case of mbed platform, we have used the mbed ticket module (derived from systick timer) to generate ticks for lvgl task handler.
For STM32 platform with lvgl, the system tick handler callback function takes care of tick count for lvgl task handler. The implementation is here: https://github.com/lvgl/lv_port_stm32f769_disco/blob/master/src/stm32f7xx_it.c#L39

So to answer, tick period is lvgl specific but in my opinion it is better to keep tick period handling separate for mbed and other platforms. Each platform may implement timer in different ways.

Reply:
Then I really don't understand the purpose of this PR.
It seems you only need a ticker at a specific frequency that calls a callback where you have to perform `lv_tick_inc(1)` .

In that case either deal with it in application code (calling HAL stuff directly) or, if you care about your project running on multiple platforms, extend the mbed_timer.cpp to be able to configure a timer that fires an interrupt which calls your desired callback (in this example it's heartbeat()):
https://github.com/analogdevicesinc/no-OS/blob/microwave/projects/microwave/src/examples/iio_admv96x1/main.c#L31

The example above uses RTC, you could use the RTC implementation on Maxim to get inspiration on how to do that for Mbed Timer.

Reply:
There could be other lvgl platform specific functions as well. I am still exploring them, so please hold off the review until I figure our those. If not, as suggested, ticker can be implemented through timer.

Reply:
@buha I have added few more supporting functions to init lvgl and related h/w.

So you need lvgl_init() function which is part of lvgl library and is purely abstracted, so no h/w dependency.
You also need touchpad and tft/display init functions, which must be vendor specific (e.g. stm displays- https://github.com/lvgl/lv_port_stm32f769_disco/tree/master/hal_stm_lvgl). These 2 functions can be part of application.

Let me know if this makes sense to you.